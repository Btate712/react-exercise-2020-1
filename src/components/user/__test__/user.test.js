import React from 'react';
import ReactDOM from 'react-dom';
import User from '../user';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

afterEach(cleanup);

test('Renders as "loading" if undefined', () => {
  const component = renderer.create(<User user={undefined} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders user if defined', () => {
  const component = renderer.create(<User user={{
    name: "Bob", 
    id: 1, 
    posts: [
      {
        title: "Post 1",
        date: "2/2/2020",
        content: "blah, blah, blah"
      },{
        title: "Post 2",
        date: "2/3/2020",
        content: "more words..."
      },{
        title: "Post 3",
        date: "2/4/2020",
        content: "getting tired of making this stuff up..."
      }
    ]
  }} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


