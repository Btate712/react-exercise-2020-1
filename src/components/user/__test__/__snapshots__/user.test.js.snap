// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders as "loading" if undefined 1`] = `
<h2>
  loading...
</h2>
`;

exports[`Renders user if defined 1`] = `
Array [
  <h1>
    User Name: 
    Bob
  </h1>,
  <h2>
    User ID: 
    1
  </h2>,
  <h2>
    Posts:
  </h2>,
  <ul>
    <li>
      Post 1
       - 
      2/2/2020
    </li>
    <li>
      Post 2
       - 
      2/3/2020
    </li>
    <li>
      Post 3
       - 
      2/4/2020
    </li>
  </ul>,
]
`;
